Traits/RedisTrait.php

```diff
@@ -29,562 +29,570 @@
-        } elseif ($redis instanceof \Predis\ClientInterface) {
-            $results = $redis->pipeline(static function ($redis) use ($generator, &$ids) {
-                foreach ($generator() as $command => $args) {
-                    $redis->{$command}(...$args);
-                    $ids[] = 'eval' === $command ? $args[2] : $args[0];
-                }
-            });
-        } elseif ($redis instanceof \RedisArray) {
-            $connections = $results = $ids = [];
-            foreach ($generator() as $command => $args) {
-                $id = 'eval' === $command ? $args[1][0] : $args[0];
-                if (!isset($connections[$h = $redis->_target($id)])) {
-                    $connections[$h] = [$redis->_instance($h), -1];
-                    $connections[$h][0]->multi(\Redis::PIPELINE);
-                }
-                $connections[$h][0]->{$command}(...$args);
-                $results[] = [$h, ++$connections[$h][1]];
-                $ids[] = $id;
-            }
-            foreach ($connections as $h => $c) {
-                $connections[$h] = $c[0]->exec();
-            }
-            foreach ($results as $k => [$h, $c]) {
-                $results[$k] = $connections[$h][$c];
-            }
-        } else {
-            $redis->multi(\Redis::PIPELINE);
-            foreach ($generator() as $command => $args) {
-                $redis->{$command}(...$args);
-                $ids[] = 'eval' === $command ? $args[1][0] : $args[0];
-            }
-            $results = $redis->exec();
-        }
-
-        if (!$redis instanceof \Predis\ClientInterface && 'eval' === $command && $redis->getLastError()) {
-            $e = new \RedisException($redis->getLastError());
-            $results = array_map(function ($v) use ($e) { return false === $v ? $e : $v; }, $results);
-        }
-
-        foreach ($ids as $k => $id) {
-            yield $id => $results[$k];
-        }
-    }
-
-    private function getHosts(): array
-    {
-        $hosts = [$this->redis];
-        if ($this->redis instanceof \Predis\ClientInterface) {
-            $connection = $this->redis->getConnection();
-            if ($connection instanceof ClusterInterface && $connection instanceof \Traversable) {
-                $hosts = [];
-                foreach ($connection as $c) {
-                    $hosts[] = new \Predis\Client($c);
-                }
-            }
-        } elseif ($this->redis instanceof \RedisArray) {
-            $hosts = [];
-            foreach ($this->redis->_hosts() as $host) {
-                $hosts[] = $this->redis->_instance($host);
-            }
-        } elseif ($this->redis instanceof RedisClusterProxy || $this->redis instanceof \RedisCluster) {
-            $hosts = [];
-            foreach ($this->redis->_masters() as $host) {
-                $hosts[] = new RedisClusterNodeProxy($host, $this->redis);
-            }
-        }
-
-        return $hosts;
-    }
+	private static $defaultConnectionOptions = [
```

Comment:
please keep the original indentation

Reply:
@nicolas-grekas https://github.com/symfony/symfony/pull/45281